dev/mcal/inc/Adc.h

```diff
@@ -0,0 +1,103 @@
+/**
+ * @file Adc.h
+ * @brief 
+ * @version 0.1
+ * @date 2024-07-20
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+#ifndef DEV_MCAL_INC_ADC_H_
+#define DEV_MCAL_INC_ADC_H_
+#include "../../mcal/inc/stm32f103xx.h"
+#define ADC_TIMEOUT (400)
+using namespace stm32::registers::adc;
+namespace stm32 {
+namespace dev   {
+namespace mcal  {
+namespace adc   {
+enum  Channel {
+        kChannel0,
+        kChannel1,
+        kChannel2,
+        kChannel3,
+        kChannel4,
+        kChannel5,
+        kChannel6,
+        kChannel7,
+        kChannel8,
+        kChannel9,
+        kChannel10,
+        kChannel11,
+        kChannel12,
+        kChannel13,
+        kChannel14,
+        kChannel15,
+        kChannel16,
+        kChannel17
+};
+enum  Alignment {
+        kRight,
+        kLeft
+};
+enum Mode {
+        kSingle = 0,
+        kContinous = 1,
+        kInjected
```

Comment:
There is no mode called `kInjected` in our board **STM32F103xxx**.
### The only available modes are:
- Single conversion Mode
- Continuous  conversion Mode
- Scan Mode
- Discontinuous Mode
[Check This article](https://deepbluembedded.com/stm32-adc-tutorial-complete-guide-with-examples/#stm32-adc-modes-of-operation)